### PR TITLE
3/rich styles

### PR DESCRIPTION
**Pull request by stuartromanek**
Builds the UI described here
https://apostrophecms.atlassian.net/jira/software/projects/A30U/boards/4?assignee=5d9f805f30387c0dbe48415e&selectedIssue=A30U-74

sample area config
```
      main: {
        type: 'area',
        options: {
          widgets: {
            '@apostrophecms/rich-text': {
              toolbar: [
                'styles',
                '|',
                'bold',
                'italic',
                'strike',
                'link',
                '|',
                'bullet_list',
                'ordered_list',
                '|',
                'blockquote',
                'code_block',
                '|',
                'horizontal_rule',
                '|',
                'undo',
                'redo'
              ],
              styles: [
                { tag: 'p', label: 'Paragraph (P)' },
                { tag: 'h3', label: 'Heading 3 (H3)' },
                { tag: 'h4', label: 'Heading 4 (H4)' }
              ]
            }
          }
        }
      }
```

The default mode uses the tiptap bubble menu, which is activated when text is **highlighted**. This menu floats above your selection (working ex https://tiptap.dev/floating-menu)

In cases where this floating UI might be problematic (narrow widget, wide toolbar, etc) you can pass `menuType: 'block'` to the `'@apostrophecms/rich-text'` options object and get something closer to https://tiptap.dev/hiding-menu-bar

### CODE REVIEW

**Comment by stuartromanek** [comment]
Once this feature gets merged I will do a cleanup component renaming PR (ApostropheRichTextWidgetEditor become AposRichTextWidgetEditor, etc)

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarUser.vue
Comment (stuartromanek) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarUser.vue:57]: `no-motion` removes hover/focus movement and was generalized from an existing type, this is just an update

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerSelections.vue
Comment (stuartromanek) [inline modules/@apostrophecms/image/ui/apos/components/AposMediaManagerSelections.vue:10]: just an update

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerSelections.vue
Comment (stuartromanek) [inline modules/@apostrophecms/image/ui/apos/components/AposMediaManagerSelections.vue:31]: just an update

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerSelections.vue
Comment (stuartromanek) [inline modules/@apostrophecms/image/ui/apos/components/AposMediaManagerSelections.vue:62]: just an update

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheRichTextWidgetEditor.vue
Comment (stuartromanek) [inline modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheRichTextWidgetEditor.vue:145]: focused is too general when using the menu, wait for it to have a selection

File: modules/@apostrophecms/ui/ui/apos/components/AposContextMenuDialog.vue
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/components/AposContextMenuDialog.vue:80]: dialogs are now inside one another, modifiers should only modify their local elements

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarUser.vue
Comment (abea) [reply modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarUser.vue:57]: Idea: use `reduced-motion` to reflect `prefers-reduced-motion`? Just an idea in case it could be connected later. https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheRichTextWidgetEditor.vue
Comment (abea) [inline modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheRichTextWidgetEditor.vue:11]: `apos-theme-dark apos-context-menu__dialog--unpadded` look like they should come from options passed in. At least the unpadded one.

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheRichTextWidgetEditor.vue
Comment (abea) [inline modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheRichTextWidgetEditor.vue:39]: It doesn't look like you're using `EditorMenuBubble`

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheRichTextWidgetEditor.vue
Comment (abea) [inline modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheRichTextWidgetEditor.vue:67]: It doesn't look like you're using `EditorMenuBubble`

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheTiptapLink.vue
Comment (abea) [inline modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheTiptapLink.vue:33]: I'm not sure how I feel about this given that a whole other third theme could be applied at the top sometime later.

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheTiptapStyles.vue
Comment (abea) [inline modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheTiptapStyles.vue:11]: Will there be more than one option for each style? Why add the index?

**Review by abea — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheTiptapLink.vue
Comment (boutell) [reply modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheTiptapLink.vue:33]: This feels weird to me too but I defer to the frontend experts.

**Review by boutell — APPROVED** [review]
Approving because the only issues in play seem to be frontend ones that I don't have much of an informed opinion on.

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheRichTextWidgetEditor.vue
Comment (stuartromanek) [reply modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheRichTextWidgetEditor.vue]: Removed instances of forcing a different theme on a specific app. This will break some design consistency with the art documents but will remove this anti pattern approach

Some long term solutions discussed:
- Themes come equipped with a fuller inverse/contrast-focused color set to use in these cases.
- Editors or developers can choose a different 'in-context' theme to handle UI overlaid on their project.

Follow up ticket created https://apostrophecms.atlassian.net/jira/software/projects/A30U/boards/4/backlog?selectedIssue=A30U-587

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheTiptapLink.vue
Comment (stuartromanek) [reply modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheTiptapLink.vue]: :stinky:

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheTiptapStyles.vue
Comment (stuartromanek) [reply modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheTiptapStyles.vue]: habit, removed

**Comment by stuartromanek** [comment]
> Approving because the only issues in play seem to be frontend ones that I don't have much of an informed opinion on.

Removed this weirdness, created a ticket to address the concern properly https://apostrophecms.atlassian.net/jira/software/projects/A30U/boards/4/backlog?selectedIssue=A30U-587

**Review by localghost443 — APPROVED** [review]

**Pull request merged**
